### PR TITLE
fix(curriculum): external cocktail link in CIP

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

clear task 109 in #40749

### CODE REVIEW

**Review by huyenltnguyen — DISMISSED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md [hunk 1]
```diff
@@ -8,29 +8,7 @@ dashedName: sorting-algorithmscocktail-sort
...
-      <b>break do-while loop</b>;
-    swapped := false
-    <b>for each</b> i <b>in</b> length( A ) - 2 <b>down to</b> 0 <b>do</b>
-      <b>if</b> A[ i ] > A[ i+1 ] <b>then</b>
-        swap( A[ i ], A[ i+1 ] )
-        swapped := true;
-  <b>while</b> swapped; <i>// if no elements have been swapped,</i>
-                <i>// then the list is sorted</i>
-</pre>
+Cocktail sort works by iterating through the list, comparing adjacent elements and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md:11]:
```suggestion
Cocktail sort works by iterating through lists, comparing adjacent elements and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Review by Sembauke — DISMISSED** [review]

**Review by jeremylt — DISMISSED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md [hunk 2]
```diff
@@ -8,29 +8,7 @@ dashedName: sorting-algorithmscocktail-sort
...
-      <b>break do-while loop</b>;
-    swapped := false
-    <b>for each</b> i <b>in</b> length( A ) - 2 <b>down to</b> 0 <b>do</b>
-      <b>if</b> A[ i ] > A[ i+1 ] <b>then</b>
-        swap( A[ i ], A[ i+1 ] )
-        swapped := true;
-  <b>while</b> swapped; <i>// if no elements have been swapped,</i>
-                <i>// then the list is sorted</i>
-</pre>
+Cocktail sort works by iterating through lists, comparing adjacent elements and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.
```
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md:11]: Not necessary, but we could add in a serial comma here:

```suggestion
Cocktail sort works by iterating through lists, comparing adjacent elements, and swapping them if they’re in the wrong order. This algorithm operates by traversing through a given array in both directions.
```

Comment (scissorsneedfoodtoo) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md:11]: Also not necessary, but we may be able to combine these two sentences:

```suggestion
Cocktail sort works by iterating through lists in both directions, comparing adjacent elements, and swapping them if they’re in the wrong order.
```

**Review by scissorsneedfoodtoo — APPROVED** [review]
~~Thanks again for your help, @Sboonny.~~

~~Everything LGTM :+1: :+1:~~

**Review by scissorsneedfoodtoo — DISMISSED** [review]
Sorry I missed this earlier, @Sboonny, but I double checked and it seems like the current definition of cocktail sort is very close to the one used in this article: https://www.growingwiththeweb.com/2016/04/cocktail-sort.html

Would it be possible to update the description here? I'd be fine with going back to text that's close to the original description, but without the external links.

**Comment by Sboonny** [comment]
No worries I will redo it👌

**Review by moT01 — COMMENTED** [review]
I guess I would be fine with it like this if others are. I think this would be quite difficult to solve with the given information. I suppose any sort would solve it, but I'm not sure this explains how a cocktail sort really works. I would need to look it up. I don't think it's too much of an issue, just pointing it out. Maybe this would be one where we keep one of the links - or maybe add this one, `https://rosettacode.org/wiki/Sorting_algorithms/Cocktail_sort`

**Comment by Sboonny** [comment]
> I'm not sure this explains how a cocktail sort really works. I would need to look it up. I don't think it's too much of an issue, just pointing it out. Maybe this would be one where we keep one of the links - or maybe add this one, https://rosettacode.org/wiki/Sorting_algorithms/Cocktail_sort

I will research it more, if I can't provide sufficient description for it, I will add link back

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md [hunk 3]
```diff
@@ -8,29 +8,9 @@ dashedName: sorting-algorithmscocktail-sort
...
-    <b>for each</b> i <b>in</b> length( A ) - 2 <b>down to</b> 0 <b>do</b>
-      <b>if</b> A[ i ] > A[ i+1 ] <b>then</b>
-        swap( A[ i ], A[ i+1 ] )
-        swapped := true;
-  <b>while</b> swapped; <i>// if no elements have been swapped,</i>
-                <i>// then the list is sorted</i>
-</pre>
+The [cocktail shaker](https://rosettacode.org/wiki/Sorting_algorithms/Cocktail_sort) sort is an improvement on the Bubble Sort. While the data is an `array` of numbers, cocktail sorting them run from both sides from the array. The **firstSide** starts at `arr[0]` and the **secondSide** starts `arr[arr.length - 1]`.
+
+Cocktail sorting checks if the element `arr[0]` of the **firstSide** is bigger than the one following it, then it swaps their positions, a similar process happens to the **secondSide** it's checking whether the element `arr[arr.length - 1]` is smaller than the one behind it, then swaps their positions. This process is iterated until you get your cocktail sorted array.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md:13]: What do you think of this @Sboonny?

```suggestion
The [cocktail shaker](https://rosettacode.org/wiki/Sorting_algorithms/Cocktail_sort) sort is an improvement on the Bubble Sort. Given an array of numbers, the cocktail shaker sort will traverse the array from start to finish, moving the largest number to the end. Then, traverse the array backwards and move the smallest number to the start. It repeats these two passes, moving the next largest/smallest number to its correct position in the array until it is sorted.
```

**Review by Sembauke — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**